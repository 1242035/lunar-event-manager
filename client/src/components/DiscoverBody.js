import React from 'react';
import DiscoverTitleCard from '../components/DiscoverTitleCard';
import ConnectedDiscoverCenterCardRow from '../components/DiscoverCenterCardRow';

/**
 * Parent element for the content of discover page.
 * @param{Object} props - passed properties
 * @return{React.Component}
 */
class DiscoverBody extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <div className="io-body-content">
        <div className="container-fluid">
          <DiscoverTitleCard>Event Centers Near You</DiscoverTitleCard>
          <ConnectedDiscoverCenterCardRow type="near" showViewCenterModal={this.props.showViewCenterModal} />
          <DiscoverTitleCard>Available Event Centers</DiscoverTitleCard>
          <ConnectedDiscoverCenterCardRow showViewCenterModal={this.props.showViewCenterModal} />
        </div>
      </div>
    );
  }
}

export default DiscoverBody;
