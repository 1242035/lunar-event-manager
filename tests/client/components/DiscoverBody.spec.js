import React from 'react';
import { shallow } from 'enzyme';
import DiscoverBody from '../../../client/src/components/DiscoverBody';

describe('DiscoverBody', () => {
  let wrapper;

  it('Should maintain existing snapshot', () => {
    wrapper = shallow(<DiscoverBody />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should render connected component', () => {
    wrapper = shallow(<DiscoverBody />);
    expect(wrapper.length).toEqual(1);
  });
});
