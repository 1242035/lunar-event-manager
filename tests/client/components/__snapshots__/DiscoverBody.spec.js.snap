// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiscoverBody Should maintain existing snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DiscoverBody />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container-fluid"
      >
        <DiscoverTitleCard>
          Event Centers Near You
        </DiscoverTitleCard>
        <Connect(DiscoverCenterCardRow)
          type="near"
        />
        <DiscoverTitleCard>
          Available Event Centers
        </DiscoverTitleCard>
        <Connect(DiscoverCenterCardRow) />
      </div>,
      "className": "io-body-content",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <DiscoverTitleCard>
            Event Centers Near You
          </DiscoverTitleCard>,
          <Connect(DiscoverCenterCardRow)
            type="near"
          />,
          <DiscoverTitleCard>
            Available Event Centers
          </DiscoverTitleCard>,
          <Connect(DiscoverCenterCardRow) />,
        ],
        "className": "container-fluid",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Event Centers Near You",
          },
          "ref": null,
          "rendered": "Event Centers Near You",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "showViewCenterModal": undefined,
            "type": "near",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Available Event Centers",
          },
          "ref": null,
          "rendered": "Available Event Centers",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "showViewCenterModal": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container-fluid"
        >
          <DiscoverTitleCard>
            Event Centers Near You
          </DiscoverTitleCard>
          <Connect(DiscoverCenterCardRow)
            type="near"
          />
          <DiscoverTitleCard>
            Available Event Centers
          </DiscoverTitleCard>
          <Connect(DiscoverCenterCardRow) />
        </div>,
        "className": "io-body-content",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <DiscoverTitleCard>
              Event Centers Near You
            </DiscoverTitleCard>,
            <Connect(DiscoverCenterCardRow)
              type="near"
            />,
            <DiscoverTitleCard>
              Available Event Centers
            </DiscoverTitleCard>,
            <Connect(DiscoverCenterCardRow) />,
          ],
          "className": "container-fluid",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Event Centers Near You",
            },
            "ref": null,
            "rendered": "Event Centers Near You",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "showViewCenterModal": undefined,
              "type": "near",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Available Event Centers",
            },
            "ref": null,
            "rendered": "Available Event Centers",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "showViewCenterModal": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
